manim/animation/animation.py

```diff
@@ -26,29 +23,29 @@
 class Animation:
     def __init__(
         self,
-        mobject: Mobject,
+        mobject: Union[Mobject, None],
```

Comment:
From https://docs.python.org/3/library/typing.html#typing.Union:
> You can use Optional[X] as a shorthand for Union[X, None].

Reply:
> @cobordism : can you wait with merging until #1338 is a bit more formulated?
> Then we can come back to this pr and know exactly how we want to have things consistent.

do you think that helps? Rather than merging and then continuing the work? After all, the type hints still require a lot of work overall... and at least this PR makes a start at reducing mypy errors.

On the whole, I'm somewhat ambivalent. It's up to you.